x-pack/plugins/apm/public/hooks/use_local_storage.ts

```diff
@@ -5,38 +5,36 @@
  * 2.0.
  */
 
-import { useState, useEffect } from 'react';
+import { useState, useEffect, useMemo } from 'react';
 
 export function useLocalStorage<T>(key: string, defaultValue: T) {
-  const [item, setItem] = useState<T>(getFromStorage(key, defaultValue));
+  const [storageUpdate, setStorageUpdate] = useState(0);
 
-  const updateFromStorage = () => {
-    const storedItem = getFromStorage(key, defaultValue);
-    setItem(storedItem);
-  };
+  const item = useMemo(() => {
+    return getFromStorage(key, defaultValue);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [key, storageUpdate, defaultValue]);
```

Comment:
Maybe a short comment that the `storageUpdate` dependency addresses the race condition